### PR TITLE
WT-6306 Mark page dirty before force evicting HS pages with deleted items

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

**Comment by sulabhM** [comment]
@agorrod , does this look like in the right direction.

File: src/btree/bt_curnext.c [hunk 1]
```diff
@@ -688,9 +688,17 @@ __wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
...
          * care not to force eviction of pages that are genuinely empty, in new trees.
+         *
+         * A visible stop timestamp could have been treated as a tombstone and accounted in the
+         * deleted count. Such a page might not have any new updates and be clean, but could benefit
+         * from reconciliation getting rid of the obsolete content. Hence mark the page dirty to
+         * force it through reconciliation.
          */
         if (page != NULL && (cbt->page_deleted_count > WT_BTREE_DELETE_THRESHOLD ||
                               (newpage && cbt->page_deleted_count > 0))) {
+            if (page->modify == NULL)
```
Comment (agorrod) [inline src/btree/bt_curnext.c:699]: I don't think you need a null check - it's built into modify_init - similarly below.

File: src/include/cursor.i [hunk 2]
```diff
@@ -210,8 +211,19 @@ __cursor_reset(WT_CURSOR_BTREE *cbt)
...
      * we release it.
+     *
+     * A visible stop timestamp could have been treated as a tombstone and accounted in the deleted
+     * count. Such a page might not have any new updates and be clean, but could benefit from
+     * reconciliation getting rid of the obsolete content. Hence mark the page dirty to force it
+     * through reconciliation.
      */
+    page = cbt->ref->page;
     if (cbt->page_deleted_count > WT_BTREE_DELETE_THRESHOLD) {
+        if (page != NULL) {
```
Comment (agorrod) [inline src/include/cursor.i:222]: I'm surprised the page can be NULL here - did you see that it was, or were you just being safe?

Comment (sulabhM) [reply src/include/cursor.i:222]: We check for page not being null in `__wt_btcur_next` and `__wt_btcur_next`, so I thought there is a possibility that reset might see a NULL page too.

Comment (sulabhM) [reply src/include/cursor.i]: Actually evict soon dereferences page without checking, so page has to exist. I have cleaned it up.

**Pull request merged**